updating/updating_a_cluster/update-using-custom-machine-config-pools.adoc

```diff
@@ -12,21 +12,42 @@ WARNING: This assembly has been moved into a subdirectory for 4.14+. Changes to
+* You want a more controlled rollout of worker node updates to ensure that mission-critical applications stay available during the whole update, even if the update process causes your applications to fail.
+* You want to update a small subset of worker nodes, evaluate cluster and workload health over a period of time, and then update the remaining nodes.
+* You want to fit worker node updates, which often require a host reboot, into smaller defined maintenance windows when it is not possible to take a large maintenance window to update the entire cluster at one time.
 
-For example, if you have a cluster with 100 nodes with 10% excess capacity, maintenance windows that must not exceed 4 hours, and you know that it takes no longer than 8 minutes to drain and reboot a worker node, you can leverage MCPs to meet your goals. For example, you could define four MCPs, named *workerpool-canary*, *workerpool-A*, *workerpool-B*, and *workerpool-C*, with 10, 30, 30, and 30 nodes respectively.
+In these scenarios, you can create multiple custom machine config pools (MCPs) to prevent certain worker nodes from updating when you update the cluster.
+After the rest of the cluster is updated, you can update those worker nodes in batches at appropriate times.
 
-During your first maintenance window, you would pause the MCP for *workerpool-A*, *workerpool-B*, and *workerpool-C*, then initiate the cluster update. This updates components that run on top of {product-title} and the 10 nodes which are members of the *workerpool-canary* MCP, because that pool was not paused. The other three MCPs are not updated, because they were paused.  If for some reason, you determine that your cluster or workload health was negatively affected by the *workerpool-canary* update, you would then cordon and drain all nodes in that pool while still maintaining sufficient capacity until you have diagnosed the problem. When everything is working as expected, you would then evaluate the cluster and workload health before deciding to unpause, and thus update, *workerpool-A*, *workerpool-B*, and *workerpool-C* in succession during each additional maintenance window.
+[id="example_{context}"]
+== Example Canary update strategy
 
-While managing worker node updates using custom MCPs provides flexibility, it can be a time-consuming process that requires you execute multiple commands. This complexity can result in errors that can affect the entire cluster. It is recommended that you carefully consider your organizational needs and carefully plan the implemention of the process before you start.
+The following example describes a canary update strategy where you have a cluster with 100 nodes with 10% excess capacity, you have maintenance windows that must not exceed 4 hours, and you know that it takes no longer than 8 minutes to drain and reboot a worker node.
```

Comment:
@skopacz1 How did you come up with 8 minutes for each node drain out as an example?

Reply:
in https://docs.openshift.com/container-platform/4.14/updating/understanding_updates/understanding-openshift-update-duration.html we have 
```
The time to complete this process depends on several factors including the node and infrastructure configuration. This process might take 5 or more minutes to complete per node.
```

Reply:
This is existing content from the current page, as can be seen [here](https://docs.openshift.com/container-platform/4.14/updating/updating_a_cluster/update-using-custom-machine-config-pools.html#:~:text=For%20example%2C%20if%20you%20have%20a%20cluster%20with%20100%20nodes%20with%2010%25%20excess%20capacity%2C%20maintenance%20windows%20that%20must%20not%20exceed%204%20hours%2C%20and%20you%20know%20that%20it%20takes%20no%20longer%20than%208%20minutes%20to%20drain%20and%20reboot%20a%20worker%20node%2C).

I'm guessing that the person who added this content originally just chose an example number for the sake of demonstrating how they could plan a canary rollout strategy given certain factors, and I don't think they were suggesting users create an expectation of their nodes having the same exact drain time. 

If you'd like to replace that value and help me update the rest of the example here, I'd be happy to do that. Or I can add some note about these only being example values and how your cluster may perform differently.

Reply:
> I can add some note about these only being example values and how your cluster may perform differently.

Agree with ^. The time to drain a node may vary depending on how much user workload there is.